src/misc/lv_event.c

```diff
@@ -183,6 +183,152 @@ lv_event_code_t lv_event_get_code(lv_event_t * e)
     return e->code & ~LV_EVENT_PREPROCESS;
 }
 
+const char * lv_event_code_get_name(lv_event_code_t code)
+{
+    switch(code & ~LV_EVENT_PREPROCESS) {
```

Comment:
See my main comment

```suggestion
    code &= ~LV_EVENT_PREPROCESS;
    switch(code) {
```

So that the compiler realizes an enum is being switched on and the enumeration exhaustion check works.

Reply:
This PR: https://github.com/lvgl/lvgl/pull/7430 already supports getting the event code name and can be modified based on the latest mainline code.

Reply:
Oh I see. Thanks FASTSHIFT.

@pnb990 please merge/rebase your branch on the latest `master`. After doing so, please add your `code & ~LV_EVENT_PREPROCESS` logic to it and then add the feedback from my other comment. Thanks!

Reply:
@liamHowatt Fixed: https://github.com/lvgl/lvgl/pull/7468